tokio/src/util/usdt/macos.rs

```diff
@@ -0,0 +1,168 @@
+//! for arg in args {
+//!     link_name.push('$');
+//!     link_name.push_str(hex::encode(arg));
+//! }
+//! ```
+//!
+//! A list of common types:
+//! * `*const core::ffi::c_char` -> `char *` -> `63686172202a`
+//! * `usize` -> `uintptr_t` -> `75696e747074725f74`
+//! * `u64` -> `uint64_t` -> `75696e7436345f74`
+//! * `u32` -> `uint32_t` -> `75696e7433325f74`
+//! * `u8` -> `uint8_t` -> `75696e74385f74`
+//!
+//! Since these are regular functions, the arguments must be passed in
+
+use std::arch::global_asm;
+
+cfg_rt!(
+    #[inline(always)]
+    pub(super) fn task_details(task_id: u64, name: &str, file: &str, line: u32, col: u32) {
+        unsafe extern "C" {
+            #[link_name = "__dtrace_isenabled$tokio$task__details$v1"]
+            fn task_details_enabled() -> i32;
+
+            #[link_name = "__dtrace_probe$tokio$task__details$v1$75696e7436345f74$63686172202a$63686172202a$75696e7433325f74$75696e7433325f74"]
+            #[cold]
+            fn __task_details(
+                task_id: u64,
+                name: *const core::ffi::c_char,
+                file: *const core::ffi::c_char,
+                line: u32,
+                col: u32,
+            );
+        }
+
+        if unsafe { task_details_enabled() } != 0 {
+            // add nul bytes
+            let name0 = [name.as_bytes(), b"\0"].concat();
+            let file0 = [file.as_bytes(), b"\0"].concat();
+
+            unsafe {
+                __task_details(
+                    task_id,
+                    name0.as_ptr() as *const std::ffi::c_char,
+                    file0.as_ptr() as *const std::ffi::c_char,
+                    line,
+                    col,
+                );
+            }
+        }
+    }
+
+    #[inline(always)]
+    pub(super) fn task_start(task_id: u64, spawned: u8, size: usize, original_size: usize) {
+        extern "C" {
+            #[link_name = "__dtrace_probe$tokio$task__start$v1$75696e7436345f74$75696e74385f74$75696e747074725f74$75696e747074725f74"]
+            fn __task_start(task_id: u64, kind: u8, size: usize, original_size: usize);
+        }
+
+        unsafe {
+            __task_start(task_id, spawned, size, original_size);
+        }
+    }
+
+    #[inline(always)]
+    pub(super) fn task_poll_start(task_id: u64) {
+        extern "C" {
+            #[link_name = "__dtrace_probe$tokio$task__poll__start$v1$75696e7436345f74"]
+            fn __task_poll_start(task_id: core::ffi::c_ulonglong);
+        }
+
+        unsafe { __task_poll_start(task_id) }
+    }
+
+    #[inline(always)]
+    pub(super) fn task_poll_end(task_id: u64) {
+        extern "C" {
+            #[link_name = "__dtrace_probe$tokio$task__poll__end$v1$75696e7436345f74"]
+            fn __task_poll_end(task_id: core::ffi::c_ulonglong);
+        }
+
+        unsafe { __task_poll_end(task_id) }
+    }
+
+    #[inline(always)]
+    pub(crate) fn task_terminate(task_id: u64, reason: u8) {
```

Comment:
```suggestion
    pub(super) fn task_terminate(task_id: u64, reason: u8) {
```
Does it need to be crate-private ? AFAIS it is used only locally (util/usdt/mod.rs)